tests/zip.rs

```diff
@@ -73,3 +73,21 @@ fn zip_eq_panic2() {
 
     zip_eq(&a, &b).count();
 }
+
+#[should_panic]
```

Comment:
Ah! I just checked (`ripgrep should_panic`) we don't really have `should_panic` tests which is valuable. But maybe a quickcheck test instead in `quick.rs`, something like
```rust
#[should_panic]
fn zip_eq_panics(a: Vec<u8>, b: Vec<u8>) -> TestResult {
    if a.len() == b.len() { return TestResult::discard(); }
    a.zip_eq(b).for_each(|_| {});
}
```
unless quickcheck tests do not like to panic. I don't know I didn't try.

Reply:
works perfectly well :) have replaced the custom tests with a couple quick tests